src/common/vector_operations/boolean_operators.cpp

```diff
@@ -35,15 +35,15 @@ template <class OP> static void templated_boolean_nullmask(Vector &left, Vector
 		right.Orrify(count, rdata);
 
 		result.vector_type = VectorType::FLAT_VECTOR;
-		auto left_data = (bool *)ldata.data;
-		auto right_data = (bool *)rdata.data;
+		auto left_data = (uint8_t *)ldata.data; // we use uint8 to avoid load of gunk bools
+		auto right_data = (uint8_t *)rdata.data;
 		auto result_data = FlatVector::GetData<bool>(result);
 		auto &result_mask = FlatVector::Nullmask(result);
 		if (ldata.nullmask->any() || rdata.nullmask->any()) {
 			for (idx_t i = 0; i < count; i++) {
 				auto lidx = ldata.sel->get_index(i);
 				auto ridx = rdata.sel->get_index(i);
-				bool is_null = OP::Operation(left_data[lidx], right_data[ridx], (*ldata.nullmask)[lidx],
+				bool is_null = OP::Operation(left_data[lidx] > 0, right_data[ridx] > 0, (*ldata.nullmask)[lidx],
```

Comment:
@Mytherin its not pretty but it avoids the load of gunky booleans. Any objections?